sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -875,7 +878,10 @@ public virtual Task<Response<ConfigurationSetting>> SetReadOnlyAsync(string key,
         /// <param name="isReadOnly">If true, the <see cref="ConfigurationSetting"/> will be set to read only in the configuration store. If false, it will be set to read write.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
         public virtual Response<ConfigurationSetting> SetReadOnly(string key, bool isReadOnly, CancellationToken cancellationToken = default)
-            => SetReadOnly(key, label: default, isReadOnly, cancellationToken);
+        {
+            Argument.AssertNotNullOrEmpty(key, nameof(key));
+            return SetReadOnlyAsync(key, default, default, isReadOnly, false, cancellationToken).EnsureCompleted();
```

Comment:
Why is EnsureCompleted used here when it is not used by any of the other methods in the configuration client?

Reply:
`EnsureCompleted` extension method for `ValueTask` has been added to `Azure.Core` several months after we have released AzConfig. So instead of having common code that handles both sync and async code paths we have code duplicates. I can add a separate task to refactor it.